### ISSUE TITLE
log_metrics step= conflicts with host run's global step counter (embedded/attach_existing use case)

### ISSUE THREAD

**Issue by rishabh-16** [gepa-ai/gepa#284]
## Problem

When GEPA is embedded inside a larger training loop that already has an active wandb run, GEPA's `log_metrics(metrics, step=gepa_state.i + 1)` conflicts with the host's global step counter.

### What happens

1. The host training loop logs metrics at its own step counter: `wandb.log({...}, step=100)`, `step=101`, `step=102`, etc.
2. GEPA optimization is triggered periodically (e.g., every N training steps).
3. GEPA internally calls `experiment_tracker.log_metrics(metrics, step=gepa_state.i + 1)` — logging at `step=1`, `step=2`, `step=3`, etc.
4. wandb's global step counter is **monotonically increasing**. When GEPA logs at `step=1` after the host already logged at `step=100`, wandb either drops the data or overwrites earlier entries.

### Result

Only a single datapoint (typically the first or last) from each GEPA optimization round is visible in wandb charts, even though multiple candidates were accepted and logged. This affects all metrics logged via `log_metrics` — `val_program_average`, `best_score_on_valset`, `iteration`, etc.

### Reproduction

```python
import wandb
from gepa.optimize_anything import optimize_anything, GEPAConfig, TrackingConfig

wandb.init(project="test")

# Host logs at high step numbers
for step in range(100):
    wandb.log({"host/loss": 1.0 / (step + 1)}, step=step)

# GEPA runs with attach_existing — logs at step=1, 2, 3...
# These conflict with the host's step counter
config = GEPAConfig()
config.tracking = TrackingConfig(
    use_wandb=True,
    wandb_attach_existing=True,
    key_prefix="gepa_round_100/",
)
result = optimize_anything(
    seed_candidate={"prompt": "..."},
    evaluator=my_evaluator,
    dataset=my_dataset,
    valset=my_valset,
    config=config,
)

# Only 1 of GEPA's ~5 accepted candidate metrics will be visible in wandb
```

### Note on tables

`log_table()` has a separate but related issue: each call creates a new `wandb.Table` with a single row and logs it under the same key, which **replaces** the previous table rather than appending. This means only the last accepted candidate's table row survives per GEPA optimization round.

---

## Proposed solution: `wandb.define_metric` with a custom x-axis

Instead of using wandb's global `step=` parameter, GEPA should declare a custom x-axis for its metrics using [`wandb.define_metric`](https://docs.wandb.ai/ref/python/run/#define_metric):

```python
wandb.define_metric(f"{key_prefix}*", step_metric=f"{key_prefix}iteration")
```

Then include the iteration as a regular metric value in each `log_metrics` call instead of passing it as `step=`:

```python
# Current (broken in embedded mode):
wandb.log({"gepa_round_100/val_program_average": 0.5}, step=3)

# Proposed:
wandb.log({
    "gepa_round_100/val_program_average": 0.5,
    "gepa_round_100/iteration": 3,
})
```

### Why this works

- **Embedded mode**: GEPA's metrics get their own x-axis (`{prefix}iteration`), completely decoupled from the host's global step. No collision, no backwards-step issues.
- **Standalone mode**: When `key_prefix` is empty, either skip `define_metric` (metrics use global step as before) or use `define_metric("*", step_metric="iteration")` — both preserve current behavior.
- **Multiple GEPA rounds**: Each round gets an independent x-axis (e.g., `gepa_round_100/iteration` vs `gepa_round_200/iteration`).
- **Multi-callsite logging**: All `log_metrics` / `log_table` calls within the same iteration include `{prefix}iteration: N` in their data. wandb groups them by the custom step metric. Multiple calls at the same iteration merge correctly.

### Implementation sketch

In `ExperimentTracker`:

1. **`start_run()` / `initialize()`**: If `key_prefix` is set and wandb is active:
   ```python
   wandb.define_metric(f"{self.key_prefix}*", step_metric=f"{self.key_prefix}iteration")
   ```

2. **`log_metrics(metrics, step=)`**: Inject the iteration into the metrics dict, drop the `step=` argument:
   ```python
   metrics_with_iter = {self._p(k): v for k, v in metrics.items()}
   metrics_with_iter[self._p("iteration")] = step
   wandb.log(metrics_with_iter)  # no step= parameter
   ```

3. **`log_table()`**: Unchanged — already uses `commit=False`.

### For the table overwrite issue

A separate fix is needed: accumulate table rows across iterations and log the full table each time, or use wandb's incremental table logging.
